platform-api/che-core-api-ssh/src/main/java/org/eclipse/che/api/ssh/server/SshManager.java

```diff
@@ -0,0 +1,88 @@
+package org.eclipse.che.api.ssh.server;
+
+import com.jcraft.jsch.JSch;
+import com.jcraft.jsch.JSchException;
+import com.jcraft.jsch.KeyPair;
+
+import org.eclipse.che.api.core.ApiException;
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.ssh.server.model.impl.SshPairImpl;
+import org.eclipse.che.api.ssh.server.model.impl.UsersSshPairImpl;
+import org.eclipse.che.api.ssh.server.spi.SshDao;
+import org.eclipse.che.api.ssh.shared.model.SshPair;
+
+import javax.inject.Inject;
+import java.io.ByteArrayOutputStream;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Facade for Ssh related operations.
+ *
+ * @author Sergii Leschenko
+ */
+public class SshManager {
+    private final JSch   genJsch;
+    private final SshDao sshDao;
+
+    @Inject
+    public SshManager(SshDao sshDao) {
+        this.sshDao = sshDao;
+        this.genJsch = new JSch();
+    }
+
+    public SshPairImpl generatePair(String owner, String serviceName, String name) throws ServerException, ConflictException {
+        KeyPair keyPair;
+        try {
+            keyPair = KeyPair.genKeyPair(genJsch, 2, 2048);
+        } catch (JSchException e) {
+            throw new ServerException("Failed to generate ssh pair.", e);
+        }
+
+        ByteArrayOutputStream privateBuff = new ByteArrayOutputStream();
+        keyPair.writePrivateKey(privateBuff);
+
+        ByteArrayOutputStream publicBuff = new ByteArrayOutputStream();
+        keyPair.writePublicKey(publicBuff, null);
+
+        final UsersSshPairImpl generatedSshPair = new UsersSshPairImpl(owner,
+                                                                       serviceName,
+                                                                       name,
+                                                                       publicBuff.toString(),
+                                                                       privateBuff.toString());
+        sshDao.create(generatedSshPair);
+        return generatedSshPair;
+    }
+
+    public void createPair(String owner, SshPair sshPair) throws ServerException, ConflictException {
+        sshDao.create(new UsersSshPairImpl(owner,
+                                           sshPair.getService(),
+                                           sshPair.getName(),
+                                           sshPair.getPublicKey(),
+                                           sshPair.getPrivateKey()));
+    }
+
+    public SshPairImpl getPair(String owner, String serviceName, String name) throws NotFoundException, ServerException {
+        return sshDao.get(owner, serviceName, name);
+    }
+
+    public List<SshPair> getPairs(String owner, String service) throws ServerException {
+        return sshDao.get(owner, service).stream().collect(Collectors.toList());
```

Comment:
Why `List<SshPair>`? It is kind of weird stuff as all the methods return `Impl` while this doesn't.
What about changing the method return type to the  type returned by the dao, then you can remove this useless collect